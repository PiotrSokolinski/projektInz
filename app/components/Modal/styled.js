import styled from 'styled-components'
import { WindowClose } from 'styled-icons/fa-regular/WindowClose'
import { motion } from 'framer-motion'
import { Colors } from '../../themes/index'

export const Close = styled(WindowClose)`
  color: ${Colors.black};
  cursor: pointer;
`

export const Background = styled(motion.div)`
  z-index: 102;
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.4);
  display: flex;
  justify-content: center;
  align-items: center;
`

export const Container = styled(motion.div)`
  border-radius: 5px;
  max-width: ${({ maxWidth }) => maxWidth}px;
  max-height: 100%;
  width: 100%;
  min-width: 300px;
  min-height: 200px;
  background-color: ${Colors.white};
  padding: 20px;
  display: flex;
  overflow: auto;
  flex-direction: column;
`

export const Header = styled.div`
  display: flex;
  padding-bottom: 30px;
  width: 100%;
`
export const Title = styled.h1`
  color: ${Colors.black};
  font-weight: bold;
  font-size: 24px;
  line-height: 36px;
  letter-spacing: 0.02em;
  flex: 1;
`

export const Description = styled.h3`
  color: ${Colors.black};
  font-size: 13px;
  line-height: 140%;
  letter-spacing: 0.02em;
  flex: 1;
  ${props => props.isConfirmation && 'margin-top: 30px;'}
`

export const CloseButton = styled.button`
  background-color: transparent;
  border: none;
  cursor: pointer;
  display: flex;
  height: 15px;
  margin-left: 25px;
  outline: none;
  padding: 0;
  width: 15px;
`
export const CloseIcon = styled.img`
  flex: 1;
  object-fit: contain;
`

export const Content = styled.div`
  flex: 1;
`

export const TextContainer = styled.div`
  flex: 1;
`
