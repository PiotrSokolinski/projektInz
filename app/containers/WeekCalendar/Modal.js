/* eslint-disable */

import React, { useState } from 'react'
import PropTypes from 'prop-types'

const Modal = ({ onRemove, start, end, onSave }) => {
  const [value, setValue] = useState('')
  const handleRemove = () => {
    onRemove()
  }
  const renderText = () => {
    if (start.isSame(end, 'day')) {
      return <span>{`${start.format('Do MMM., HH:mm')} - ${end.format('HH:mm')}`}</span>
    }
    return (
      <span>{`${start.format('Do MMM.')} - ${end.format('Do MMM.')}, ${start.format('HH:mm')} - ${end.format(
        'HH:mm',
      )}`}</span>
    )
  }
  const handleSave = () => {
    onSave({
      value,
    })
  }
  return (
    <div className="customModal">
      <div className="customModal__text">{renderText()}</div>
      <input
        className="customModal__input"
        type="text"
        placeholder="Enter something"
        value={value}
        onChange={e => setValue(e.target.value)}
      />
      <button className="customModal__button" onClick={handleRemove}>
        Delete
      </button>
      <button className="customModal__button customModal__button_float_right" onClick={handleSave}>
        Save
      </button>
    </div>
  )
}

Modal.propTypes = {
  start: PropTypes.object.isRequired,
  end: PropTypes.object.isRequired,
  onRemove: PropTypes.func.isRequired,
  onSave: PropTypes.func.isRequired,
  actionType: PropTypes.string, // eslint-disable-line react/no-unused-prop-types
}

export default Modal
