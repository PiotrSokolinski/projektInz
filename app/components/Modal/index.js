/**
 *
 * Modal
 *
 */

import React, { useState, useEffect, useRef } from 'react'
import PropTypes from 'prop-types'
import { useAnimation } from 'framer-motion'
import * as Styled from './styled'

const ESC_KEY_CODE = 27

const springTransition = {
  type: 'spring',
  damping: 22,
  stiffness: 400,
}

const defaultTransition = {
  duration: 0.15,
}

const modalHiddenAnimationConfig = {
  scale: 0.5,
  transition: defaultTransition,
}
const modalVisibleAnimation = {
  scale: 1,
  transition: springTransition,
}
const backgroundHiddenAnimationConfig = {
  opacity: 0,
  transition: defaultTransition,
}
const backgroundVisibleAnimationConfig = {
  opacity: 1,
  transition: defaultTransition,
}

const Modal = ({ description, title, children, onClose, visible: shouldBeVisible, maxWidth, className }) => {
  const [renderModal, setRenderModal] = useState(false)
  const [showModal, setShowModal] = useState(false)
  const modalContainer = useRef(null)
  const modalOverlay = useRef(null)
  const containerAnimationController = useAnimation()
  const backgroundAnimationController = useAnimation()
  const handleClick = event => {
    const { target } = event
    if (
      modalOverlay.current &&
      modalContainer.current &&
      modalOverlay.current.contains(target) &&
      !modalContainer.current.contains(target)
    )
      onClose()
  }

  const handleKeyUp = event => {
    if (event.keyCode === ESC_KEY_CODE) onClose()
  }

  useEffect(() => {
    window.addEventListener('click', handleClick, false)
    window.addEventListener('keyup', handleKeyUp, false)
    return () => {
      window.removeEventListener('click', handleClick, false)
      window.removeEventListener('keyup', handleKeyUp, false)
    }
  }, [])

  useEffect(() => {
    if (shouldBeVisible) {
      if (!renderModal) {
        setRenderModal(true)
      } else if (!showModal) {
        backgroundAnimationController.start(backgroundVisibleAnimationConfig)
        containerAnimationController.start(modalVisibleAnimation)
        setShowModal(true)
      }
    } else if (renderModal) {
      if (showModal) {
        backgroundAnimationController.start(backgroundHiddenAnimationConfig)
        containerAnimationController.start(modalHiddenAnimationConfig).then(() => setRenderModal(false))
        setShowModal(false)
      }
    }
  }, [shouldBeVisible, renderModal, showModal])

  return (
    renderModal && (
      <Styled.Background
        className={className}
        ref={modalOverlay}
        id="modal-overlay"
        initial={backgroundHiddenAnimationConfig}
        animate={backgroundAnimationController}
      >
        <Styled.Container
          className="modal-container"
          maxWidth={maxWidth}
          ref={modalContainer}
          initial={modalHiddenAnimationConfig}
          animate={containerAnimationController}
        >
          <Styled.Header className="modal-header">
            <Styled.TextContainer>
              <Styled.Title className="modal-title">{title}</Styled.Title>
              <Styled.Description>{description}</Styled.Description>
            </Styled.TextContainer>
            <Styled.Close size="22" onClick={onClose} />
          </Styled.Header>
          <Styled.Content>{children}</Styled.Content>
        </Styled.Container>
      </Styled.Background>
    )
  )
}

Modal.propTypes = {
  children: PropTypes.node,
  description: PropTypes.string,
  maxWidth: PropTypes.number,
  onClose: PropTypes.func.isRequired,
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
  visible: PropTypes.bool,
}

Modal.defaultProps = {
  description: '',
  maxWidth: 500,
  visible: true,
}

export default Modal
