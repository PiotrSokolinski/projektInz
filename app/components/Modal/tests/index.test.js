/**
 *
 * Tests for Modal
 *
 */

import React from 'react'
import { render } from '@testing-library/react'

import Modal from '../index'

describe('<Modal />', () => {
  const title = 'mocked tittle'
  const content = 'mocked content text'
  const onClose = jest.fn()

  beforeEach(() => {
    onClose.mockClear()
  })

  it('Should not render modal', () => {
    const { container } = render(<Modal visible={false} title={title} onClose={onClose} />)
    expect(container).toMatchSnapshot()
  })
  it('Should render modal', () => {
    const { container } = render(<Modal visible={false} title={title} onClose={onClose} />)
    expect(container).toMatchSnapshot()
  })
  it('Should become visible', async () => {
    const { container } = render(<Modal visible title={title} onClose={onClose} />)
    expect(container).toMatchSnapshot()
    // time for animation
    await new Promise(res => setTimeout(res, 1000))
    expect(container).toMatchSnapshot()
  })
  it.skip('should invoke onClose method after button press', () => {
    const { container } = render(<Modal visible title={title} onClose={onClose} />)
    container.getElementsByTagName('img')[0].click()
    expect(onClose).toHaveBeenCalled()
  })
  it('should render children correctly', () => {
    const { container } = render(
      <Modal visible title={title} onClose={onClose}>
        {content}
      </Modal>,
    )
    expect(container).toMatchSnapshot()
  })
})
