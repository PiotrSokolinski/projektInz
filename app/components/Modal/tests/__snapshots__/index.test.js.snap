// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Should become visible 1`] = `
<div>
  <div
    class="styled__Background-sc-1gkpoto-1 crBfyX"
    id="modal-overlay"
    style="opacity: 0;"
  >
    <div
      class="modal-container styled__Container-sc-1gkpoto-2 enawXZ"
      style="transform: scale(0.5) translateZ(0);"
    >
      <div
        class="modal-header styled__Header-sc-1gkpoto-3 gtBJHt"
      >
        <div
          class="styled__TextContainer-sc-1gkpoto-9 GkMdz"
        >
          <h1
            class="modal-title styled__Title-sc-1gkpoto-4 kvaykP"
          >
            mocked tittle
          </h1>
          <h3
            class="styled__Description-sc-1gkpoto-5 jbdhCY"
          />
        </div>
        <svg
          aria-hidden="true"
          class="styled__Close-sc-1gkpoto-0 kZkAkZ StyledIconBase-sc-bdy9j4 cCvWXa"
          fill="currentColor"
          focusable="false"
          height="22"
          viewBox="0 0 512 512"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm0 394c0 3.3-2.7 6-6 6H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h404c3.3 0 6 2.7 6 6v340zM356.5 194.6L295.1 256l61.4 61.4c4.6 4.6 4.6 12.1 0 16.8l-22.3 22.3c-4.6 4.6-12.1 4.6-16.8 0L256 295.1l-61.4 61.4c-4.6 4.6-12.1 4.6-16.8 0l-22.3-22.3c-4.6-4.6-4.6-12.1 0-16.8l61.4-61.4-61.4-61.4c-4.6-4.6-4.6-12.1 0-16.8l22.3-22.3c4.6-4.6 12.1-4.6 16.8 0l61.4 61.4 61.4-61.4c4.6-4.6 12.1-4.6 16.8 0l22.3 22.3c4.7 4.6 4.7 12.1 0 16.8z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="styled__Content-sc-1gkpoto-8 cGAdLK"
      />
    </div>
  </div>
</div>
`;

exports[`<Modal /> Should become visible 2`] = `
<div>
  <div
    class="styled__Background-sc-1gkpoto-1 crBfyX"
    id="modal-overlay"
    style="opacity: 1;"
  >
    <div
      class="modal-container styled__Container-sc-1gkpoto-2 enawXZ"
      style="transform: none;"
    >
      <div
        class="modal-header styled__Header-sc-1gkpoto-3 gtBJHt"
      >
        <div
          class="styled__TextContainer-sc-1gkpoto-9 GkMdz"
        >
          <h1
            class="modal-title styled__Title-sc-1gkpoto-4 kvaykP"
          >
            mocked tittle
          </h1>
          <h3
            class="styled__Description-sc-1gkpoto-5 jbdhCY"
          />
        </div>
        <svg
          aria-hidden="true"
          class="styled__Close-sc-1gkpoto-0 kZkAkZ StyledIconBase-sc-bdy9j4 cCvWXa"
          fill="currentColor"
          focusable="false"
          height="22"
          viewBox="0 0 512 512"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm0 394c0 3.3-2.7 6-6 6H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h404c3.3 0 6 2.7 6 6v340zM356.5 194.6L295.1 256l61.4 61.4c4.6 4.6 4.6 12.1 0 16.8l-22.3 22.3c-4.6 4.6-12.1 4.6-16.8 0L256 295.1l-61.4 61.4c-4.6 4.6-12.1 4.6-16.8 0l-22.3-22.3c-4.6-4.6-4.6-12.1 0-16.8l61.4-61.4-61.4-61.4c-4.6-4.6-4.6-12.1 0-16.8l22.3-22.3c4.6-4.6 12.1-4.6 16.8 0l61.4 61.4 61.4-61.4c4.6-4.6 12.1-4.6 16.8 0l22.3 22.3c4.7 4.6 4.7 12.1 0 16.8z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="styled__Content-sc-1gkpoto-8 cGAdLK"
      />
    </div>
  </div>
</div>
`;

exports[`<Modal /> Should not render modal 1`] = `<div />`;

exports[`<Modal /> Should render modal 1`] = `<div />`;

exports[`<Modal /> should render children correctly 1`] = `
<div>
  <div
    class="styled__Background-sc-1gkpoto-1 crBfyX"
    id="modal-overlay"
    style="opacity: 0;"
  >
    <div
      class="modal-container styled__Container-sc-1gkpoto-2 enawXZ"
      style="transform: scale(0.5) translateZ(0);"
    >
      <div
        class="modal-header styled__Header-sc-1gkpoto-3 gtBJHt"
      >
        <div
          class="styled__TextContainer-sc-1gkpoto-9 GkMdz"
        >
          <h1
            class="modal-title styled__Title-sc-1gkpoto-4 kvaykP"
          >
            mocked tittle
          </h1>
          <h3
            class="styled__Description-sc-1gkpoto-5 jbdhCY"
          />
        </div>
        <svg
          aria-hidden="true"
          class="styled__Close-sc-1gkpoto-0 kZkAkZ StyledIconBase-sc-bdy9j4 cCvWXa"
          fill="currentColor"
          focusable="false"
          height="22"
          viewBox="0 0 512 512"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h416c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm0 394c0 3.3-2.7 6-6 6H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h404c3.3 0 6 2.7 6 6v340zM356.5 194.6L295.1 256l61.4 61.4c4.6 4.6 4.6 12.1 0 16.8l-22.3 22.3c-4.6 4.6-12.1 4.6-16.8 0L256 295.1l-61.4 61.4c-4.6 4.6-12.1 4.6-16.8 0l-22.3-22.3c-4.6-4.6-4.6-12.1 0-16.8l61.4-61.4-61.4-61.4c-4.6-4.6-4.6-12.1 0-16.8l22.3-22.3c4.6-4.6 12.1-4.6 16.8 0l61.4 61.4 61.4-61.4c4.6-4.6 12.1-4.6 16.8 0l22.3 22.3c4.7 4.6 4.7 12.1 0 16.8z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="styled__Content-sc-1gkpoto-8 cGAdLK"
      >
        mocked content text
      </div>
    </div>
  </div>
</div>
`;
